version: '3.8'

services:
  app:
    build: ./app
    env_file:
      - ./app/.env
    volumes:
      - ./app:/app
    ports:
      - "8000:8000"
    networks:
      - app-network
    command: python main.py
    depends_on:
      - database
      - rabbitmq

  web-proxy:
    build: ./nginx
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - app
    networks:
      - app-network

  rabbitmq:
    image: rabbitmq:3.12-management
    ports:
      - "15672:15672"
      - "5672:5672"
    volumes:
      - rabbitmq_volume:/var/lib/rabbitmq
    restart: on-failure
    env_file:
      - ./app/.env
    environment:
      - RABBITMQ_DEFAULT_USER=${RABBITMQ_USER}
      - RABBITMQ_DEFAULT_PASS=${RABBITMQ_PASSWORD}

  database:
    image: postgres:16-alpine
    container_name: myapp
    restart: unless-stopped
    volumes:
      - postgres_volume:/var/lib/postgresql/data
    networks:
      - app-network
    env_file:
      - ./app/.env
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
      

volumes:
  postgres_volume:
  rabbitmq_volume:

networks:
  app-network:
    name: app-network
    driver: bridge